src/routes/v2/authenticated/review.ts

```diff
@@ -379,16 +379,118 @@ export class ReviewsRouter {
       })
     }
 
-    // Step 3: Merge review request
-    // NOTE: We are not checking for existence of RR
+    // Step 3: Retrieve review request
+    const possibleReviewRequest = await this.reviewRequestService.getReviewRequest(
+      site,
+      requestId
+    )
+
+    if (isIsomerError(possibleReviewRequest)) {
+      return res.status(404).json({ message: possibleReviewRequest.message })
+    }
+
+    // Step 4: Merge review request
+    // NOTE: We are not checking for existence of PR
     // as the underlying Github API returns 404 if
     // the requested review could not be found.
-    await this.reviewRequestService.mergeReviewRequest(
-      siteName,
-      requestId,
-      site.id
+    await this.reviewRequestService.mergeReviewRequest(possibleReviewRequest)
```

Comment:
should we add a try catch block here to cater for unexpected network blip issues and this `mergeReviewRequest` doesnt go through?

Reply:
could i clarify what the `catch` block would do in this case? if we're just going to report the error, this behaviour might already be captured by `errorHandler`

Reply:
It was more to return a `res`. Ie, if there is an error thrown in `await this.reviewRequestService.mergeReviewRequest(possibleReviewRequest)`, I would rather return a `res.status(400).send("something went wrong")` or `res.status(500).send("something went wrong")`

Reply:
If the error is thrown, it short circuits to `errorHandler` which does exactly what you just described! 

Fact check @alexanderleegs